### PR TITLE
chore: adds and fixes e2e for content history

### LINKED ISSUE #20350: Content history "back" link doesn't work
### Required System information

Environment: development
OS: darwin-arm64
Strapi Version: 5.0.0-beta.7
Node/Yarn Version: yarn/4.0.1 npm/? node/v20.9.0 darwin arm64
Edition: Enterprise
Database: sqlite

### Describe the bug

On the "content history" page, clicking the back button in the top left of the page causes a new browser window to open to the login rather than going back

### Steps to reproduce the behavior

1. Create an entity in the content manager
2. Click on the ... menu in the top right
3. Select "Content History"
4. On the content history page, click on the "back" button in the top left
5. See that it doesn't work and instead opens a new browser window

### Expected behavior

Clicking "back" should go back

### Screenshots

image

### Code snippets

If applicable, add code samples to help explain your problem.

### PR DESCRIPTION
**Pull request by remidej**
### What does it do?

- Adds e2e tests to ensure that Content History handles missing relations properly.
- Fixes issues with the design system upgrade: links opening in a new tab, links not having the right color
- Various e2e test fixes

### Related issue(s)/PR(s)

- Tests the implementation of #20054 and #20035
- closes #20295, closes #20350

### CODE REVIEW

**Review by markkaylor — COMMENTED** [review]
Tests look good to me and are passing locally. But some of the other tests are now failing. Maybe related to the flaky tests already mentioned... 🤔

**Comment by remidej** [comment]
Yes I'm planning to come back to it after the beta and investigate the timeouts

File: tests/e2e/tests/content-manager/history.spec.ts [hunk 1]
```diff
@@ -7,40 +7,71 @@ import { waitForRestart } from '../../utils/restart';
...
+
+/**
+ * The success notification lets us make sure an action is completed before moving on.
+ * We also close it to make sure only one appears at a time, otherwise the test fails.
+ */
+const closeSuccessNotification = async (page: Page) => {
+  const successNotification = page.getByRole('alert', { name: /success/i });
+  if (await successNotification.isVisible()) {
+    await successNotification.getByRole('button', { name: /close/i }).click();
+  }
```
Comment (markkaylor) [inline tests/e2e/tests/content-manager/history.spec.ts:48]: Not positive but it looks like this util might do the same thing, or is meant to do the same thing...

https://github.com/strapi/strapi/blob/f75e3c6d67cc47c64ab37479efdbb7b43be50b78/tests/e2e/utils/shared.ts#L27-L35

File: tests/e2e/tests/content-manager/history.spec.ts [hunk 2]
```diff
@@ -7,40 +7,71 @@ import { waitForRestart } from '../../utils/restart';
...
 describeOnCondition(edition === 'EE')('History', () => {
-  test.describe('Collection Type', () => {
-    test.beforeEach(async ({ page }) => {
-      await resetDatabaseAndImportDataFromPath('with-admin.tar');
-      await resetFiles();
-      await page.goto('/admin');
-      await login({ page });
-    });
+  test.beforeEach(async ({ page }) => {
+    await resetDatabaseAndImportDataFromPath('with-admin.tar', (cts) => cts, { coreStore: false });
```
Comment (markkaylor) [inline tests/e2e/tests/content-manager/history.spec.ts:53]: I'm sure there is a reason but I'm struggling to understand why this config was needed and set to false here.

File: tests/e2e/tests/content-manager/history.spec.ts [hunk 3]
```diff
@@ -213,13 +284,6 @@ describeOnCondition(edition === 'EE')('History', () => {
   });
 
   test.describe('Single Type', () => {
-    test.beforeEach(async ({ page }) => {
-      await resetDatabaseAndImportDataFromPath('with-admin.tar');
-      await resetFiles();
-      await page.goto('/admin');
-      await login({ page });
-    });
```
Comment (markkaylor) [inline tests/e2e/tests/content-manager/history.spec.ts:221]: Why is this removed? Shouldn't we continue to reset the db and git working tree between tests to ensure the tests are isolated?

File: tests/e2e/tests/content-manager/history.spec.ts (hunk 2 above)
Comment (remidej) [reply tests/e2e/tests/content-manager/history.spec.ts:53]: Currently on v5/main the core_store isn't reset by the function between tests.

core_store is used for "configure the view changes", and for us it meant that when we have a test the deletes a field from the layout, the test that runs after that doesn't have the input.

I feel like the core_store should always be reset, but the way it was configured looked intentional, so I chose to modify it only for us via a param to avoid breaking other tests, at least for now

File: tests/e2e/tests/content-manager/history.spec.ts (hunk 3 above)
Comment (remidej) [reply tests/e2e/tests/content-manager/history.spec.ts:221]: We had the same beforeEach for single and collection types, so I moved it up to the parent describe, now it runs for everything

Comment (markkaylor) [reply tests/e2e/tests/content-manager/history.spec.ts:221]: Oh I see what you did, but is that the same thing? Are you sure that runs for each test inside each nested describe block? I thought it would just run before each nested describe but not each test inside those nested describe blocks.

Comment (remidej) [reply tests/e2e/tests/content-manager/history.spec.ts:221]: Yes I checked, it wouldn't pass otherwise. I see how it's not super intuitive, lmk if you'd prefer to revert it

Comment (markkaylor) [reply tests/e2e/tests/content-manager/history.spec.ts:221]: Ok good to know, fine by me if that's how it works :) thanks

**Review by markkaylor — APPROVED** [review]
